tempesta_fw/http.c

```diff
@@ -3329,7 +3334,7 @@ tfw_http_resp_process(TfwConn *conn, const TfwFsmData *data)
 	 * we won't resend it.
 	 */
 	bad_req = hmresp->req;
-	tfw_http_req_delist((TfwSrvConn *)conn, bad_req);
+	tfw_http_req_delist(hmresp);
```

Comment:
I think the idea was that both `req` and `srv_conn` were known already so there's no need to get/derive them again. The called function is inlined and so the values are just used in place. Perhaps, GCC optimizes this just as needed, and there's no "calculation" of these values again.

Reply:
Also, is it possible that this situation can happen in other places as well? This particular function is only called once. The unlocked version is called in all other cases.

Reply:
@keshonok Thank you for review!

In "hot path" functions we normally try to reduce number of calculations, but this sometimes makes function readability worse. Take a look at this mostly obsolete (and resolved) example. Function takes two parameters: `int f(TfwHttpReq *req, TfwHttpResp *resp)`. But in the same time we know, that we can get `req` as `resp->req` or `resp` as `req->resp`. So why the two different pointers are passed here? Is it the same `req` as saved in `resp->req` or not? 

It isn't a hot path function, it's very and very unlikely situation instead. We think about of protected servers as reliable and good servers, so we actually expect server to send a valid message, not a garbage that can't be parsed or processed. So extra calculations (if happen) doesn't hurt much.

In the same time, the function is self-sufficient, it takes care of server connection forwarding queue, and keeps it to be consistent (see https://github.com/tempesta-tech/tempesta/pull/1133#discussion_r241723435 ). 

That is why I thought that extra calculations doesn't hurt and being more explicit helps to make understanding of the function role (readability) better. But taking into account review comments, it seems to be partly done, I should rename the function and add a good comment description to mirror function role.